### PR TITLE
Enhance the documentation of `LightGBM` tuner and separate `train()` from `__init__.py`

### PR DESCRIPTION
**Pull request by nabenabe0928**
## Motivation

As it was hard to parse the necessary information from the current documentation string of LightGBMTuner, I will enhance the documentation.

## Description of the changes

I made the following changes:
1. Create a new file for the `train` function of `LightGBMTuner` to move the function from `__init__.py`,
2. Change `*args` and `**kwargs` of the `train` function to the exact variable names for more clarity and better UX, and
3. update the typing to adapt to the newer Python versions.

### CODE REVIEW

File: optuna/integration/_lightgbm_tuner/optimize.py [hunk 1]
```diff
@@ -750,9 +751,9 @@ class LightGBMTuner(_LightGBMBaseTuner):
     def __init__(
         self,
         params: dict[str, Any],
-        train_set: "lgb.Dataset",
```
Comment (contramundum53) [inline optuna/integration/_lightgbm_tuner/optimize.py:753]: Strings are specified as type because `lgb` is not imported (and thus the name `lgb` results in error) when the user doesn't have lightgbm installed. (This is controlled by `try_import()`.) Could you revert these changes?

File: optuna/integration/_lightgbm_tuner/optimize.py [hunk 2]
```diff
@@ -750,9 +751,9 @@ class LightGBMTuner(_LightGBMBaseTuner):
...
+        train_set: lgb.Dataset,
         num_boost_round: int = 1000,
-        valid_sets: "VALID_SET_TYPE" | None = None,
+        valid_sets: list[lgb.Dataset] | tuple[lgb.Dataset, ...] | lgb.Dataset | None = None,
```
Comment (contramundum53) [inline optuna/integration/_lightgbm_tuner/optimize.py:756]: Also, please stringify this type annotation.
```suggestion
        valid_sets: "list[lgb.Dataset] | tuple[lgb.Dataset, ...] | lgb.Dataset | None" = None,
```

File: optuna/integration/_lightgbm_tuner/_train.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    time_budget: int | None = None,
+    sample_size: int | None = None,
+    study: Study | None = None,
+    optuna_callbacks: list[Callable[[Study, FrozenTrial], None]] | None = None,
+    model_dir: str | None = None,
+    verbosity: int | None = None,
+    show_progress_bar: bool = True,
+    *,
+    optuna_seed: int | None = None,
+) -> lgb.Booster:
```
Comment (contramundum53) [inline optuna/integration/_lightgbm_tuner/_train.py:37]: Please stringify lgb types.

**Review by contramundum53 — APPROVED** [review]
LGTM.

**Comment by contramundum53** [comment]
@Alnusjaponica Could you review this PR?

**Review by Alnusjaponica — COMMENTED** [review]
Sorry for my delayed reply. 
I made some minor comments. PTAL.

File: optuna/integration/_lightgbm_tuner/optimize.py [hunk 4]
```diff
@@ -684,8 +684,12 @@ class LightGBMTuner(_LightGBMBaseTuner):
     /medium.com/optuna/lightgbm-tuner-new-optuna-integration-for-hyperparameter-optimization-8b709
     5e99258>`_ by `Kohei Ozaki <https://www.kaggle.com/confirm>`_, a Kaggle Grandmaster.
 
-    Arguments and keyword arguments for `lightgbm.train()
-    <https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.train.html>`_ can be passed.
+    .. note::
```
[the remark this answers is not in the document]
Comment (nabenabe0928) [reply optuna/integration/_lightgbm_tuner/optimize.py:687]: Nice catch, thanks! (I did not notice)

File: optuna/integration/_lightgbm_tuner/_train.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
+from __future__ import annotations
+
+from typing import Any
+from typing import Callable
```
[the remark this answers is not in the document]
Comment (nabenabe0928) [reply optuna/integration/_lightgbm_tuner/_train.py:4]: @c-bata If I remember correctly, you mentioned that we would like to somehow remove `abc` at some points, but does this also apply to this as well?
Just in case, I would to make sure.

[the remark this answers is not in the document]
Comment (HideakiImamura) [reply optuna/integration/_lightgbm_tuner/_train.py:4]: I think we should follow the PRP584. As in https://github.com/optuna/optuna/issues/4508, we change the import like this.

While I can't be certain and this is just a conjecture, I believe the recommendation to avoid the abc module likely refers to when utilizing polymorphism. Previously in Optuna, polymorphism was used by defining abstract classes with abc.ABCMeta, but recently, we've been using polymorphism through duck typing using typing.Protocol. See https://peps.python.org/pep-0544/ for more details.

**Review by Alnusjaponica — APPROVED** [review]
LGTM.
